### PR TITLE
feat: Pubsub sink 

### PR DESCRIPTION
**Pull request by turtleDev**
This PR adds support for GCP PubSub Sink (#77) to Raccoon.

# Configuration
This change introduces the following new configurations:

* `PUBLISHER_TYPE` configures which publisher to use. 
  * supported values: `kafka`, `pubsub`
  * default: `kafka`
* `PUBLISHER_PUBSUB_PROJECT_ID` project ID of the target GCP project. (required)
* `PUBLISHER_PUBSUB_CREDENTIALS` (required, but can also alternatively use `GOOGLE_APPLICATION_CREDENTIALS`)
* `PUBLISHER_PUBSUB_TOPIC_AUTOCREATE` controls whether unknown topics should be created (defaults to `false`)
* `PUBLISHER_PUBSUB_TOPIC_RETENTION_MS` defines the message retention policy for topics created by Raccon. Valid values must be between 10 minutes and 31 days (see [docs](https://cloud.google.com/pubsub/docs/replay-overview#topic_message_retention)) 
* `PUBLISHER_PUBSUB_PUBLISH_DELAY_THRESHOLD_MS` controls the time delay before messages are published (defaults to 10ms)
* `PUBLISHER_PUBSUB_PUBLISH_COUNT_THRESHOLD` specifies the internal buffer length (defaults to 100 messages) 
* `PUBLISHER_PUBSUB_PUBLISH_BYTE_THRESHOLD` specifies the internal buffer size (defaults to ~1MiB)
* `PUBLISHER_PUBSUB_PUBLISH_TIMEOUT_MS` specifies how long raccoon tries to publish a batch of messages before giving up. (defaults to 1 minute)

### CODE REVIEW

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -5,22 +5,27 @@ on:
...
-      - name: Install Protoc
-        uses: arduino/setup-protoc@v1
       - name: Checkout repo
         uses: actions/checkout@v2
-      - name: Setup Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: "1.18"
+      - name: setup integration environment
+        uses: ./.github/workflows/integration-test-setup
```
Comment (ravisuhag) [inline .github/workflows/integration-test.yaml:14]: @turtleDev Let's inline this. Any reason to split this workflow?

File: app/server.go [hunk 2]
```diff
@@ -3,44 +3,52 @@ package app
...
 	workerPool.StartWorkers()
-	go kPublisher.ReportStats()
+
 	go reportProcMetrics()
-	go shutDownServer(ctx, cancel, httpServices, bufferChannel, workerPool, kPublisher)
+	go shutDownServer(ctx, cancel, httpServices, bufferChannel, workerPool, publisher)
 }
 
-func shutDownServer(ctx context.Context, cancel context.CancelFunc, httpServices services.Services, bufferChannel chan collection.CollectRequest, workerPool *worker.Pool, kp *publisher.Kafka) {
+func shutDownServer(ctx context.Context, cancel context.CancelFunc, httpServices services.Services, bufferChannel chan collection.CollectRequest, workerPool *worker.Pool, kp Publisher) {
```
Comment (ravisuhag) [inline app/server.go:51]: @turtleDev Let's update short var as well `kp`

File: app/server.go [hunk 3]
```diff
@@ -3,44 +3,52 @@ package app
...
 	"context"
 	"fmt"
+	"io"
 	"os"
 	"os/signal"
 	"runtime"
 	"syscall"
 	"time"
 
+	"cloud.google.com/go/pubsub"
```
Comment (ravisuhag) [inline app/server.go:13]: @turtleDev what is the need of this dependency here?

File: publisher/pubsub.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
+package publisher
```
Comment (ravisuhag) [inline publisher/pubsub.go:1]: @turtleDev let's create a dedicated package for each publisher within publisher folder.

File: .github/workflows/integration-test.yaml (hunk 1 above)
Comment (turtleDev) [reply .github/workflows/integration-test.yaml:14]: This action contains the common steps necessary for running integration tests for kafka and pubsub.
This was introduced to reduce redundant workflow steps within our testing pipelines.

File: app/server.go (hunk 2 above)
Comment (turtleDev) [reply app/server.go]: renamed to `pub`.

File: app/server.go (hunk 3 above)
Comment (turtleDev) [reply app/server.go:13]: `publisher.NewPubSub()` doesn't create the pubsub client by itself. It leaves that responsiblity to the client code (in this case, `initPublisher()`)

File: publisher/pubsub.go (hunk 4 above)
Comment (turtleDev) [reply publisher/pubsub.go]: done.

**Pull request merged**